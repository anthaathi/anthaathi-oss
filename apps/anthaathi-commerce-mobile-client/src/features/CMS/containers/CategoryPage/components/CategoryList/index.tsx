import {
  ScrollView,
  Text,
  ImageBackground,
  View,
  TouchableOpacity,
  Image,
} from 'react-native';
import React from 'react';
import {CategoryPageComponentType} from '../../../../types/common';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';

export interface CategoryProps {
  title: string;
  title_ar: string;
  key: string;
  image: string;
}

export interface CategoryListProps {
  list: CategoryProps[];
  onPress?: (key: string) => void;
}

const ItemRenderer = ({
  item,
  itemWidth,
  onPress,
}: {
  item: CategoryProps;
  itemWidth: string;
  onPress: (key: string) => void;
}) => {
  return (
    <TouchableOpacity
      onPress={() => onPress(item.key)}
      style={{height: 180, width: itemWidth, margin: 10, borderRadius: 4}}>
      <Image
        source={{
          uri: item.image,
        }}
        style={{
          position: 'absolute',
          height: '100%' as never,
          width: '100%' as never,
          borderRadius: 4,
        }}
      />
      <Text
        style={{
          marginTop: 70,
          textAlign: 'center',
          color: '#000',
          fontSize: 18,
          fontWeight: '700',
          backgroundColor: '#fff',
          paddingVertical: 5,
          marginHorizontal: 20,
        }}>
        {item.title}
      </Text>
    </TouchableOpacity>
  );
};

const CategoryList = (props: CategoryListProps) => {
  const itemWidth = useResponsiveValue(['44%', '30%', '22%', '22%']);

  return (
    <ScrollView testID="categoryList">
      <View style={{flexDirection: 'row', flexWrap: 'wrap'}}>
        {props.list.map((data: CategoryProps) => (
          <ItemRenderer
            item={data}
            key={data.key}
            itemWidth={itemWidth}
            onPress={props.onPress || (() => {})}
          />
        ))}
      </View>
    </ScrollView>
  );
};

export default CategoryList;

export const CategoryListCMSInput = {
  _component: CategoryPageComponentType.CategoryList,
  component: CategoryList,
};
