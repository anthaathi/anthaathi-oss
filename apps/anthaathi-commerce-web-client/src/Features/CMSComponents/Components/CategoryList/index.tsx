import { useStyletron } from '@anthaathi/solid-styletron';
import { Link } from '@solidjs/router';
import { For } from 'solid-js';
import { Grid } from '~/Features/Core/Components/Grid';
import { Img } from '~/Features/Core/Components/Image';

export interface CategoryDetailsProps {
  id: number;
  title_ar: string;
  title: string;
  image: string;
  key: string;
  href: string;
}

export interface CategoryListProps {
  title?: string;
  items: CategoryDetailsProps[];
  handlePress?: (key: string) => void;
}

export function CategoryList(props: CategoryListProps) {
  const [css, $theme] = useStyletron();
  return (
    <div>
      {props.title && (
        <h5
          class={css([
            $theme.typography.HeadingMedium,
            {
              marginTop: $theme.sizing.scale600,
              marginBottom: $theme.sizing.scale600,
              fontWeight: 'bold',
              color: '#000',
            },
          ])}
        >
          {props.title}
        </h5>
      )}
      <Grid
        $override={{
          Root: {
            style: {
              rowGap: '8px',
              gridGap: '15px',
            },
          },
        }}
        columns={[2, 3, 4, 5, 6]}
      >
        <For each={props.items}>
          {(item) => <CategoryDetails item={item} />}
        </For>
      </Grid>
    </div>
  );
}

const CategoryDetails = ({ item }: { item: CategoryDetailsProps }) => {
  const [css, $theme] = useStyletron();
  return (
    <Link
      href={item.href}
      class={css({
        textDecoration: 'none',
        alignItems: 'center',
        display: 'flex',
        flexDirection: 'column',
      })}
    >
      <Img
        src={item.image}
        $override={{
          Root: {
            $style: {
              margin: '0px',
              objectFit: 'cover',
              // [$theme.mediaQuery?.md || '']: {
              width: '180px',
              height: '180px',
              // },
              // width: '120px',
              // height: '120px',
              borderRadius: '50%',
              ':hover': {
                animation: 'fade-in-show .4s',
                transform: 'scale(1.03)',
              },
            },
          },
        }}
      />
      <h5
        class={css([
          $theme.typography.ParagraphLarge,
          {
            marginTop: $theme.sizing.scale600,
            marginBottom: $theme.sizing.scale1000,
            fontWeight: 'bold',
            color: '#000',
          },
        ])}
      >
        {item.title}
      </h5>
    </Link>
  );
};
