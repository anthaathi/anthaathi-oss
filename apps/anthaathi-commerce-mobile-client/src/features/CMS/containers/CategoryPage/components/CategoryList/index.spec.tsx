import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import CategoryList from './index';

describe('CategoryList', () => {
  it('should render the item', function () {
    const temp = render(
      <IntlProvider locale="en-US" messages={locale}>
        <ThemeProvider>
          <CategoryList
            list={[
              {
                title: 'test',
                title_ar: 'test',
                key: 'test',
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              },
            ]}
          />
        </ThemeProvider>
      </IntlProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('categoryList')).toBeTruthy();
  });
});
