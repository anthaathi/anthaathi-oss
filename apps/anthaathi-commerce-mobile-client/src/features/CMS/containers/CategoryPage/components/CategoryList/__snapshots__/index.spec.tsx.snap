// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render the item 1`] = `
<RCTScrollView
  testID="categoryList"
>
  <View>
    <View
      style={
        {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderRadius": 4,
            "height": 180,
            "margin": 10,
            "opacity": 1,
            "width": "30%",
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
            }
          }
          style={
            {
              "borderRadius": 4,
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <Text
          style={
            {
              "backgroundColor": "#fff",
              "color": "#000",
              "fontSize": 18,
              "fontWeight": "700",
              "marginHorizontal": 20,
              "marginTop": 70,
              "paddingVertical": 5,
              "textAlign": "center",
            }
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
